File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchTypeUrls.java
package openfoodfacts.github.scrachx.openfood.utils;

import java.util.HashMap;

import openfoodfacts.github.scrachx.openfood.BuildConfig;

import static openfoodfacts.github.scrachx.openfood.utils.SearchType.ALLERGEN;
import static openfoodfacts.github.scrachx.openfood.utils.SearchType.EMB;
import static openfoodfacts.github.scrachx.openfood.utils.SearchType.TRACE;

public class SearchTypeUrls {
    private static final HashMap<String, String> URLS = new HashMap<>();

    private SearchTypeUrls() {

    }

    static {
        URLS.put(ALLERGEN, BuildConfig.OFWEBSITE + "allergens/");
        URLS.put(EMB, BuildConfig.OFWEBSITE + "packager-code/");
        URLS.put(TRACE, BuildConfig.OFWEBSITE + "trace/");
    }

    public static String getUrl(@SearchType String type) {
        return URLS.get(type);
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchTypeUrls.kt
package openfoodfacts.github.scrachx.openfood.utils

import openfoodfacts.github.scrachx.openfood.BuildConfig
import openfoodfacts.github.scrachx.openfood.utils.SearchType

object SearchTypeUrls {
    private val URLS = mapOf(
            SearchType.ALLERGEN to "${BuildConfig.OFWEBSITE}allergens/",
            SearchType.EMB to "${BuildConfig.OFWEBSITE}packager-code/",
            SearchType.TRACE to "${BuildConfig.OFWEBSITE}trace/"
    )

    fun getUrl(type: SearchType) = URLS[type]
}